### ISSUE TITLE
watchteleboy hangs at startup

### ISSUE THREAD

**Issue by frankongithub18** [reduzent/watchteleboy#30]
It looks like watchteleboy v2.0 does not work anymore. Available channels are shown with watchteleboy -l but nothing happens when i select them with -c <channel_name>.
Without cli parameter watchteleboy hangs directly.

**Comment by reduzent** [comment]
Does this work:

```
watchteleboy -c arte --print-url
```

I know that doesn't help you much, but playback still works for me. Unfortunately, the script doesn't provide many diagnostic capabilities (yet), so it is hard for me to tell, what is going. I'll have to figure out how to make it print more useful messages instead of just hanging.

**Comment by frankongithub18** [comment]
Ok, i am not familiar with python but i know other programming languages. Can I debug the script with pdb watchteleboy?
Actually i've installed watchteleboy through the ppa. The dependencies should be met and I have checked them manually, everything looks fine.

**Comment by reduzent** [comment]
Has it worked before? 

I'd certainly would give pdb a try. I haven't used it myself and also I am not a very seasoned Python programmer. I'm still exploring things as I go with watchteleboy. I'll try to familiarize myself with pdb when I find time. It looks like a useful tool.

**Comment by reduzent** [comment]
BTW: I was asking about the `--print-url` test because this interacts only with the teleboy API, but doesn't yet download anything. It might give us a coarse clue where the problem might be.

**Comment by frankongithub18** [comment]
When I execute the command 
`watchteleboy -c arte --print-url`
I'll get an URL back to the manifest
`http://zh2-5-dash-live.zahs.tv/DE_HD_arte/manifest.mpd?z32=MF2WI2LPL5RW6ZDFMNZT2YLBMMTGG43JMQ6TCNKGGY3EEMBSIM4EENJRGA2DCLJSGA4TMMJZIY2UKM2CIU3DANKFEZWWC6DSMF2GKPJVGAYDAJTNNFXHEYLUMU6TEMJQEZYHEZLGMVZHEZLEL5WGC3THOVQWOZJ5MRSSM43JM46TGMJTMM3GMNZTGY2WKYJYGY4TAYRRMUZGMMBZMVQTQMJWGBSTMMZGOVZWK4S7NFSD25DFNRSWE33ZL43DOMZQGI4TKNBW`

**Comment by reduzent** [comment]
OK, that means the hanging happens in the download code and not in the API code. Internally, when playing back a stream, the downloaded data is passed through pipes (one for video and a separate one for audio) to mpv. You could try to record (`-r`) a short video to check, whether the hanging happens in the piping or the downloading itself.

Try something like this:

```
watchteleboy -r -c arte -t 21:00 -d 60
```

You should end up with a mkv file of duration=1min. If this works, then the problem is located somewhere in the part were data is passed to pipes or with mpv subprocess.

**Comment by frankongithub18** [comment]
ah recording works this way as you describe... 
`watchteleboy -r -c arte -t 22:00 -d 60`
so has Live TV another handling for processing streams?

**Comment by reduzent** [comment]
No, recording and playback is pretty similar. However, recording writes data to a files. Playback writes the data to named pipes and mpv is started in a subprocess and reads from those pipes. All that stuff is in the class WatchTeleboyPlayer in wt/classes.py. After selecting a channel, WatchTeleboyPlayer creates two instances of WatchTeleboyStreamHandler, one for video and for audio. Those perform the actual download. mpv is then basically started like this:

```
mpv --audio-file=audio_fifo video_fifo
```
Or that is at least how it is supposed to work. Do you see a bunch of randomly named files in ~/.watchteleboy ?

**Comment by frankongithub18** [comment]
Found out the solution. I'm using Ubuntu 18.04 LTS and mpv is on version 0.27.2 by default.
Now I've added the ppa from mpv and updated the application. After this everything is working like a charm :-)

```
sudo add-apt-repository ppa:mc3man/mpv-tests
sudo apt-get update
sudo apt-get upgrade mpv
```
Many thanks for your help

**Comment by reduzent** [comment]
Thanks for the report.  I only now realize that I'm using my locally compiled version of mpv and I probably should test my stuff with the mpv that comes with Ubuntu. I'll definitely look into that.

**Comment by reduzent** [comment]
Reminder for myself: Make sure script runs with packaged mpv version!

**Comment by reduzent** [comment]
Ok, I got this fixed with  940cb1c8dbbaaa. It seems, only very recent versions of mpv understand '1GiB', older versions want plain interger numbers.

Though this specific issue is fixed,  it appears that the way mpv is called currently hides any failure thereof. watchteleboy should detect premature exit of mpv and also print the error message. I open a separate issue for that.
